pkg/function/backup_data_all.go

```diff
@@ -62,13 +66,19 @@ func init() {
 
 var _ kanister.Func = (*backupDataAllFunc)(nil)
 
-type backupDataAllFunc struct{}
+type backupDataAllFunc struct {
+	progressPercent string
+}
 
 func (*backupDataAllFunc) Name() string {
 	return BackupDataAllFuncName
 }
 
-func (*backupDataAllFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+func (b *backupDataAllFunc) Exec(ctx context.Context, tp param.TemplateParams, args map[string]interface{}) (map[string]interface{}, error) {
+	// Set progress percent
+	b.progressPercent = progress.StartedPercent
+	defer func() { b.progressPercent = progress.CompletedPercent }()
```

Comment:
@viveksinghggits in case of failure, the phase progress is set to nil. Here is an example from tests where out of 3, first phase completed but second phase fails
```
status:                                                                                                                                                                                                     
  actions:                                                                                                                                                                                                            
  - artifacts:                                                                                                                                                                                                        
      operatorBackup:                                                                                                                                                                                                 
        keyValue:                                                                                                                                                                                                     
          path: '{{ .Phases.phase3.Output.path }}'                                                                                                                                                                    
    blueprint: operator-blueprint                                                                                                                                                                                     
    deferPhase:                                                                                                                                                                                                       
      name: ""                                                                                                                                                                                                        
      progress: {}                                                                                                                                                                                                    
      state: ""                                                                                                                                                                                                       
    name: backup                                                                                                                                                                                                      
    object:                                                                                                                                                                                                           
      apiVersion: v1                                                                                                                                                                                                  
      group: apps                                                                                                                                                                                                     
      kind: statefulset                                                                                                                                                                                               
      name: mysql                                                                                                                                                                                                     
      namespace: mysql                                                                                                                                                                                                
      resource: statefulsets                                                                                                                                                                                          
    phases:                                                                                                                                                                                                           
    - name: phase1                                                                                                                                                                                                    
      output:                                                                                                                                                                                                         
        path: /operator-backups/2023-05-22T07-32-36/backup.tgz                                                                                                                                                        
      progress:                                                                                                                                                                                                       
        lastTransitionTime: "2023-05-22T07:32:54Z"                                                                                                                                                                    
        progressPercent: "100"                                                                                                                                                                                         
      state: complete                                                                                                                                                                                                 
    - name: phase2                                                                                                                                                                                                    
      progress: {}                                                                                                                                                                                                    
      state: failed                                                                                                                                                                                                   
    - name: phase3                                                                                                                                                                                                    
      progress: {}                                                                                                                                                                                                    
      state: pending                                                                                                                                                                                                  
  error:                                                                                                                                                                                                              
    message: "Failed while waiting for Pod kanister-job-sk2bk to complete: Pod failed                                                                                                                                 
      or did not transition into complete state: Running backup\
     .
     .
  progress:                                                                                                                                                             
    lastTransitionTime: "2023-05-22T07:32:54Z"                                          
    percentCompleted: "33"                                                           
  state: failed        
```